Service/InstallationService.php

```diff
@@ -3,785 +3,894 @@
+        }
 
-            foreach ($installers->files() as $installer) {
-                $this->handleInstaller($installer);
+        return true;
+    }//end addToObjects()
+    
+    /**
+     * Handles schemas of a certain type.
+     *
+     * @param string $type The type of the object
+     * @param array $schemas The schemas to handle
+     *
+     * @return void
+     */
+    private function handleObjectType(string $type, array $schemas): void
+    {
+        foreach ($schemas as $schema) {
+            $object = $this->handleObject($type, $schema);
+            if ($object === null) {
+                continue;
             }
-        } else {
-            $this->logger->debug('No Installation folder found for plugin '.$bundle);
-            $this->io->writeln('No Installation folder found');
+            
+            // Save it to the database.
+            $this->entityManager->persist($object);
         }
 
-        $this->io->success('All Done');
-        $this->logger->debug('All Done installing plugin '.$bundle);
-
-        return Command::SUCCESS;
-    }
+    }//end handleObjectType();
 
-    public function update(string $bundle, string $data)
+    /**
+     * Create an object bases on a type and a schema (the object as an array).
+     *
+     * This function breaks complexity rules, but since a switch is the most effective way of doing it a design decision was made to allow it
+     *
+     * @param string $type   The type of the object
+     * @param array  $schema The object as an array
+     *
+     * @return object|null
+     */
+    private function handleObject(string $type, array $schema): ?object
     {
-        $this->io->writeln([
-            'Common Gateway Bundle Updater',
-            '============',
-            '',
-        ]);
-
-        return Command::SUCCESS;
-    }
+        // Only base we need it the assumption that on object isn't valid until we made is so.
+        $object = null;
+
+        // For security reasons we define allowed resources.
+        $allowedCoreObjects = [
+            'https://docs.commongateway.nl/schemas/Action.schema.json',
+            'https://docs.commongateway.nl/schemas/Application.schema.json',
+            'https://docs.commongateway.nl/schemas/CollectionEntity.schema.json',
+            'https://docs.commongateway.nl/schemas/Cronjob.schema.json',
+            'https://docs.commongateway.nl/schemas/DashboardCard.schema.json',
+            'https://docs.commongateway.nl/schemas/Endpoint.schema.json',
+            'https://docs.commongateway.nl/schemas/Entity.schema.json',
+            'https://docs.commongateway.nl/schemas/Gateway.schema.json',
+            'https://docs.commongateway.nl/schemas/Mapping.schema.json',
+            'https://docs.commongateway.nl/schemas/Organization.schema.json',
+            'https://docs.commongateway.nl/schemas/SecurityGroup.schema.json',
+//                'https://docs.commongateway.nl/schemas/User.schema.json',
```

Comment:
I'm not sure if we want this eventually.

Reply:
Same here, but we might